frontend/src/components/website/adding_master/StepAddDomain.vue

```diff
@@ -0,0 +1,127 @@
+                    </VCardText>
+                    <VSwitch
+                        v-model="single_page"
+                        label="Single page app"
+                        color="#3C57DE"
+                        inset
+                    />
+                </VForm>
+                <VBtn
+                    large
+                    class="white--text mt-4"
+                    color="#3C57DE"
+                    @click="onGoToNextStep"
+                >
+                    Get Tracking Info
+                </VBtn>
+            </VContainer>
+        </VCard>
+    </VContainer>
+</template>
+
+<script>
+    import {mapActions, mapGetters} from 'vuex';
+    import {GET_NEW_WEBSITE} from "@/store/modules/website/types/getters";
+    import {SAVE_NEW_WEBSITE, SET_DOMAIN, SET_SPA} from "@/store/modules/website/types/actions";
+
+    const domainRegex = /(https?:\/\/)?(www\.)?[-a-zA-Z0-9@:%._~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_.~#()?&//=]*)/;
+
+    export default {
+        name: 'StepAddDomain',
+        props: {
+            errorMessage: {
+                type: String,
+                required: false,
+                default: ''
+            }
+        },
+        data() {
+            return {
+                domain: '',
+                valid: false,
+                single_page: false,
+                domainRules: [
+                    v => !!v || 'Website domain is required',
+                    v => domainRegex.test(v) || 'Domain must be correct'
+                ],
+            }
+        },
+        computed: {
+            ...mapGetters('website', {
+                newWebsite: GET_NEW_WEBSITE
+            }),
+            errorText() {
+                return this.errorMessage;
+            }
+        },
+        created () {
+            this.domain = this.newWebsite.domain;
+            this.single_page = this.newWebsite.single_page;
+        },
+        methods: {
+            ...mapActions('website', {
+                setDomain: SET_DOMAIN,
+                setSinglePage: SET_SPA,
+                saveNewSite: SAVE_NEW_WEBSITE
+            }),
+            onGoToNextStep () {
+                if (this.$refs.form.validate()) {
+                    this.setDomain(this.domain).then(() => {
+                        this.setSinglePage(this.single_page).then(() => {
+                            this.saveNewSite().then(() => {
+                                this.$router.push({name: 'add_websites_step_3'});
+                            })
+                                .catch((res) => {
+                                    this.onError(res.errors);
+                                });
+                        })
+                    });
+                }
+            },
+            onError (errors) {
+                if (errors.name) {
+
+                    this.$router.push({name: 'add_websites_step_1', params: {errorMessage: errors.name}});
+
+                } else if (errors.domain) {
```

Comment:
avoid using else if where it's possible, much cleaner implementation where `if` has empty return like
```
if (errors.name) {
....
    return;
}

if (errors.domain) {
.....
    return;
}

//etc.
```